# HG changeset patch
# User Rik <rik@octave.org>
# Date 1448208576 28800
#      Sun Nov 22 08:09:36 2015 -0800
# Node ID 771d9fab5c24af6b67423d51420bea5ba07b77ff
# Parent  83611b387bc5f66e78c3f304fbb07f405d5f354e
Stop assertion crash when 'echo on all' used (bug #45209).

* pt-pr-code.cc (visit_no_op_command): When end-of-function encountered, check
that indent level is greater than 1 before decreasing indent to avoid negative
values which trigger assertion crash in indent().

diff --git a/libinterp/parse-tree/pt-pr-code.cc b/libinterp/parse-tree/pt-pr-code.cc
--- a/libinterp/parse-tree/pt-pr-code.cc
+++ b/libinterp/parse-tree/pt-pr-code.cc
@@ -714,17 +714,17 @@ tree_print_code::visit_multi_assignment 
     rhs->accept (*this);
 
   print_parens (expr, ")");
 }
 
 void
 tree_print_code::visit_no_op_command (tree_no_op_command& cmd)
 {
-  if (cmd.is_end_of_fcn_or_script ())
+  if (cmd.is_end_of_fcn_or_script () && curr_print_indent_level > 1)
     decrement_indent_level ();
 
   indent ();
 
   os << cmd.original_command ();
 }
 
 void
